lib/wallet/test/unit/Cardano/Wallet/Api/TypesSpec.hs

```diff
@@ -941,18 +945,18 @@ instance FromJSON SchemaApiErrorCode where
 -------------------------------------------------------------------------------}
 
 -- Dummy instances
-instance EncodeAddress ('Testnet 0) where
+instance {-# INCOHERENT #-} EncodeAddress ('Testnet 0) where
     encodeAddress = const "<addr>"
 
-instance DecodeAddress ('Testnet 0) where
+instance {-# INCOHERENT #-} DecodeAddress ('Testnet 0) where
```

Comment:
I'm a bit curious: why do the "real" instances for {Decode,Encode}Address and {Decode,Encode}SharedAddress` (defined in `Cardano.Wallet.Shelley.Compatibility`) cause tests to fail in `Api.TypesSpec`?

If we were to simply delete these dummy instances, how much work would it be to adjust the test suite so that it uses the real instances?

Reply:
Created and linked to: https://input-output.atlassian.net/browse/ADP-2302

Btw, it seems `OVERLAPPING` was enough, which already made it to master through some other PR. But I added TODOs with links anyway.